pvlib/spa_c_files/README.md

```diff
@@ -1,34 +1,38 @@
 README
 ------
 
-NREL provides a C implementation of the solar position algorithm 
-described in 
-[Reda, I.; Andreas, A. (2003). Solar Position Algorithm for Solar Radiation Applications. 55 pp.; NREL Report No. TP-560-34302](
-http://www.nrel.gov/docs/fy08osti/34302.pdf).
+NREL provides a C implementation of the solar position algorithm described in
+[Reda, I.; Andreas, A. (2003). Solar Position Algorithm for Solar Radiation Applications. 55 pp.; NREL Report No. TP-560-34302](http://www.nrel.gov/docs/fy08osti/34302.pdf).
 
 This folder contains the files required to make NREL's C code accessible
 to the ``pvlib-python`` package. We use the Cython package to wrap NREL's SPA 
 implementation. 
 
-** Due to licensing issues, you must download the NREL C files from their 
-[website](http://www.nrel.gov/midc/spa) **
-
-Download the ``spa.c`` and ``spa.h`` files from NREL, 
-and copy them into the ``pvlib/spa_c_files`` directory. 
+** Due to licensing issues, the [SPA C files](http://www.nrel.gov/midc/spa) can
+_not_ be included in the pvlib-python distribution. The SPA C files will be
+downloaded when you build the Python extension. By using this module you agree
+to the NREL license in SPA_NOTICE. **
```

Comment:
I'm concerned about adding this feature. Can we really say this and wash our hands of the license issue? I would be surprised. I'd believe we're in the clear if we pulled the license text, printed it, and waited for the user to input ``y``.

Also not clear to me how this is supposed to work (or not work) with the spa related code in the root setup.py

Reply:
(by this feature I mean the auto-download capability)

Reply:
> I'm concerned about adding this feature. Can we really say this and wash our hands of the license issue? I would be surprised.

but the LICENSED code is not being distributed with pvlib, so I don't see the concern?

>I'd believe we're in the clear if we pulled the license text, printed it, and waited for the user to input `y`.

I guess we can do that

>Also not clear to me how this is supposed to work (or not work) with the spa related code in the root `setup.py`

AFAIK there's no difference, everything is exactly the same way it was before. The user has to follow the instructions in the README. If they want to use the SPA C-files, then they need to somehow enter this folder and run

    python setup.py build_ext --inplace

it's the only way to build it, and by doing this act, then implicitly agree to the license. If they don't want to use the SPA C-files they do nothing. AFAIK the rest of pvlib still works exactly the way it always did, what's different?

Reply:
>but the LICENSED code is not being distributed with pvlib, so I don't see the concern?

Maybe. But I don't want NREL to think "Holmgren, Mikofski, and the rest of those ^&*( pvlib-ers snaked their way around our license."

If we support automatic download, I'd also want a user to be required to input the form information rather than it being supplied as "pvlib-python" etc. NREL is presumably asking for this information from each user in order to justify funding to DOE. Putting a placeholder there doesn't seem right, regardless of whether we agree with NREL's strategy here or not. 

>AFAIK there's no difference, everything is exactly the same way it was before.

The root setup.py compiles the extension if the files were detected during a ``pip install .`` or ``python setup.py``. Or at least it used to -- it's been years since I tried. Perhaps it's best to remove the capability of the root setup.py to do anything with this extension.

Reply:
great idea, okay, it now asks for users name, and sends that, if no name, then it fails with some ugly message which contains the contents of spa.c which says something like:
>Name must be atleast 2 valid characters.

Reply:
It looks like the [root `setup.py` checks if the files are present](https://github.com/pvlib/pvlib-python/blob/master/setup.py#L87-L96) so I think it will run okay.

It looks like the CI's had no problems building it either.

Also I just checked it on my PC, and it also works as expected:
* if no `spa.c` or `spa.h` then you get this:
    >WARNING: spa_c files not detected. See installation instructions for more information.
* if yes `spa.c` or `spa.h` then it builds the extension.